### ISSUE TITLE
`declaration-block-trailing-semicolon` reads a declaration that closes no block

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#326]
### What minimal example or steps are needed to reproduce the bug?

`declaration-block-trailing-semicolon` walks every declaration of the file and asks nothing about where that declaration stands, so a declaration standing on the root — on no block at all — is read as the one closing its block and is reported.

| syntax | option | code | what comes back |
|---|---|---|---|
| `postcss-scss` | `"never"` | `$var: pink;` | `$var: pink` |
| `postcss-scss` | `"always"` | `$var: pink` | `$var: pink;` |
| plain CSS | `"never"` | `color: pink;` | `color: pink` |

The walk over at-rules, three lines above, asks the question and bails: `if (atRule.parent === root …) return`.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-scss",
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/declaration-block-trailing-semicolon": "never"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.scss" --config myconfig.json --fix
```

### What did you expect to happen?

A rule named for what closes a declaration block says nothing about a declaration that closes none. A Sass variable written at the top of a file is a statement of the file, not the last declaration of a block, and the semicolon behind it is the one Sass asks for rather than a trailing one.

### What actually happened?

The semicolon is taken away under `"never"` and written under `"always"`, in every syntax, wherever a declaration stands on the root.

### What is the cause?

`root.walkDecls` in [`declaration-block-trailing-semicolon/index.js`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/rules/declaration-block-trailing-semicolon/index.js) asks only whether the parent is a Sass map and whether the declaration is the last node of it. `root.walkAtRules`, right above it, asks `atRule.parent === root` first. The two walks disagree about what a block is.

### Do you have a proposal to fix the bug?

Ask the same question in both walks. Whether the answer should be the same for a stylesheet Stylelint opened as an inline style attribute is worth settling in the same breath: `isInlineStyleAttribute` already tells such a root apart, and the declarations of one do close something.

The branch closing [#217](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/217) widens the reach rather than opening it: a root-level declaration with a comment behind it becomes reachable there, while the rows above hold on `main` today.
